### ISSUE TITLE
Breaking Change In 0.3.2?

### ISSUE THREAD

**Issue by JimLynchCodes** [typestack/class-transformer#578]
Hey there!

I have been using this class-transformer library in my NestJs project to transform incoming stringified data to the proper types...

I had created this custom decorator that is supposed to take an incoming hex string of a Mongodb ObjectId and convert it to an actual ObjectId:

```
import { Transform } from 'class-transformer';
import { ObjectId } from 'mongodb';
export function TransformObjectId(): (target: any, key: string) => void {
  const transformToClassFn = Transform((it) => new ObjectId(it), { toClassOnly: true });
  const transformFromClassFn = Transform((it: ObjectId) => it.toHexString(), { toPlainOnly: true });
  return function (target: any, key: string) {
    transformToClassFn(target, key);
    transformFromClassFn(target, key);
  }
}
```

It was working for me fine until I upgraded from 0.3.1 to 0.3.2

Now, with 0.3.2 I see this error/warning:

```
(alias) class ObjectId
import ObjectId
A class representation of the BSON ObjectId type.
Argument of type '(it: ObjectId) => string' is not assignable to parameter of type '(params: TransformFnParams) => any'.
  Types of parameters 'it' and 'params' are incompatible.
    Type 'TransformFnParams' is not assignable to type 'ObjectId'.ts(2345)
Peek Problem (Alt+F8)
No quick fixes available
```

Does anyone know why I am seeing this? Is there a quick fix I can do to my code so that it will work again with the new lib version?

Also, what is the reasoning for not using major versions at all and pushing breaking changes as patch updates? 😅

**Comment by mopcweb** [comment]
Hi, 

spent 1.5 days trying to find why it was broken in production.

finally found - it is related to updating of [@Transform](https://github.com/typestack/class-transformer#advanced-usage) decorator callback function, which previously has another [signature](https://github.com/typestack/class-transformer/tree/master#advanced-usage), so there is a breaking change in __patch__ version.

@NoNameProvided could you please revert changes and provide them in new __major__ version?

thanks

**Comment by NoNameProvided** [comment]
`0.4.0` has been released, `0.3.2` has been deprecated with a warning message: 
> This release contains a breaking change from 0.3.1 but otherwise functions properly. Please read the changelog on Github to learn more.

Closing this as solved.
